specs/date-picker/date-picker.md

```diff
@@ -0,0 +1,285 @@
+# Date Picker
+
+## Process
+
+- [ ] **Checkpoint 1** - Schedule a spec review meeting to cover the Overview section before proceeding to work on the Design section. Be sure to include relevant stakeholders such as members of the tools team.
+
+- [ ] **Checkpoint 2** - Schedule a spec review meeting before beginning the implementation work. Be sure to invite relevant stakeholders such as design system owners.
+
+- [ ] **Checkpoint 3** - Submit the component PR and schedule a component code review meeting or present the component in one of our Engineering Open Mic sessions. Address feedback from the review, obtain final approvals through GitHub, and merge.
+
+## Overview
+
+As defined by the W3C:
+>The date state represents a control for setting the element's value to a string representing a specific date.
+>
+
+### Background
+
+This is a community requested component.
+
+### Use Cases
+
+- John is booking a room online and needs to enter a date of arrival. Clicking on the calendar icon in the date field, brings up a calendar where John can select a check-in date.
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+  
+### Features
+
+- **Date validation:** Verifies that the entered string is a valid date.
+
+- **Calendar view:** Opens a calendar view for selecting a date.
+
+- **Time view:** Opens time selection view with select boxes for selecting hour, minute and meridian.
+
+- **Month view:** Opens a month selection view. Used for quickly changing the calendar view or for selecting a month when type="month".
+
+- **Year view:** Opens a year selection view. Used for quickly changing the calendar view, month select view or for selecting a year when type="year".
+
+- **Localization:** Formats that date for the local market. MM--DD-YYYY, DD--MM-YYYY, etc. It supports country, language, numbering type and calendar type.
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing the component. Would we need to make any breaking changes in order to achieve this component's goals?*
+
+- Many different calendar types, even within the same market.
```

Comment:
Do you have an idea of how much code supporting the different calendar types in one control will be? My general concern is bloating out a single component w/ myriad support options and loading un-necessary code for markets and use-cases that don't need it. If the code to support them is small it is likely a non-issue, but if not, perhaps we can brainstorm some ways to support the myriad calendars in a way that still keeps individual implementations lightweight.

Reply:
I reduced it to what I believe to be a minimum viable product. I'm using the built-in internationalization functions of the browser for getting dates, month names, etc. This will cover all of the markets that use the Gregorian calendars. But I think there will be problems with markets using other calendars like Buddhist, Islamic, Hebrew, etc. There are 13 defined calendar types in the browser. But it's more complicated than that. A specific calendar can be different in different markets. For example, there are different leap years with long and short months for the Buddhist calendar in Cambodia than the Buddhist calendar in Burma. 

The only way that I might be able to differentiate is to walk backwards and forward from the current date looking for month and year changes. This might be slow, so I left it as a future nice to have feature.